### PR TITLE
tests(rpc): add filter tests

### PR DESCRIPTION
**Pull request by danburck**
## Description

This PR adds basic integration tests for testing JSON RPC eth filters on a ethermint/geth cluster. In the future we should add more complex testing cases.

Closes https://linear.app/evmos/issue/ENG-642/integration-test-json-rpc-filters

### CODE REVIEW

**Review by fedekunze — APPROVED** [review]
LGTM, can you fix the linter errors?

**Comment by danburck** [comment]
@facs95 @ramacarlucho @yihuang seems like the test_get_proof is failing even though this PR doesn't touch the types.

The last two merged commits passed:
* https://github.com/evmos/ethermint/pull/1236
* https://github.com/evmos/ethermint/pull/1237

Any idea why it's failing?

File: tests/integration_tests/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+For linux:
+
+```
+sh <(curl -L https://nixos.org/nix/install) --no-daemon
+```
+
+## Run Local
+
+First time run (can take a while):
```
Comment (yihuang) [inline tests/integration_tests/README.md:29]: Instruct the user to add binary cache as well? could speed up a lot.

Comment (yihuang) [reply tests/integration_tests/README.md:29]:
```
## Enable Binary Cache

```
$ nix-env -iA cachix -f https://cachix.org/api/v1/install
$ cachix use ethermint
```

**Comment by yihuang** [comment]
> Any idea why it's failing?

Have no idea right now, didn't reproduce locally, maybe you try rerun, it'll success 🤔

**Comment by yihuang** [comment]
I think the priority test failure is due to the dropping of the base fee, which causes the actual tx priority not as expected when sent, I'll update some numbers to make the drop effect less significant.

**Comment by yihuang** [comment]
> I think the priority test failure is due to the dropping of the base fee, which causes the actual tx priority not as expected when sent, I'll update some numbers to make the drop effect less significant.

https://github.com/evmos/ethermint/pull/1238

**Comment by yihuang** [comment]
> seems like the test_get_proof is failing even though this PR doesn't touch the types

https://github.com/evmos/ethermint/pull/1239

File: tests/integration_tests/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# RPC Integration tests
+
+The RPC integration test suite uses nix and python to send identical queries to both an Ethermint and a [Geth](https://github.com/ethereum/go-ethereum) client and compare the responses. It allows for quickly assessing the compatibility with Geth.
```
Comment (facs95) [inline tests/integration_tests/README.md:3]: Thank you for adding this Readme! 🙏 We were lacking this instructions. I would just suggest changing this description as the integration tests are not only concern about Geth but basically any configuration. Maybe something like this?

```
The RPC integration test suite uses nix for reproducible and configurable
builds allowing to run integration tests using python web3 library against 
different Ethermint and Geth clients with multiple configurations.
```

**Review by facs95 — APPROVED** [review]

File: tests/integration_tests/README.md (hunk 2 above)
Comment (danburck) [reply tests/integration_tests/README.md]: Good point, thank you!

**Review by yihuang — APPROVED** [review]

**Pull request merged**